tor/core/mentions.py

```diff
@@ -35,11 +38,20 @@ def process_mention(mention, r, tor, redis_server, config):
         parent.title = 'Unknown Content'
     else:
         # this is a post.
-        parent = r.submission(id=clean_id(mention.parent_id))
+        parent = submission
         parent_permalink = parent.permalink
         # format that sucker so it looks right in the template.
         parent.title = '"' + parent.title + '"'
 
+    # Ignore requests made by the OP of content or the OP of the submission
+    if mention.author.id == parent.author.id or \
+            mention.author.id == submission.author.id:
```

Comment:
I think we're definitely on the right track here. Could this be simplified to just `if mention.author == submission.author`? Also, there's another function that may be of use here. I have a helper function, located in tor.helpers.reddit_ids called get_parent_post_id -- it's expensive, but it's entire goal in life is to take any comment object and return the top submission ID. Is that something that might be able to clean this up a little?